In [2]:
import matplotlib.pyplot as plt
import pandas as pd
from collections import defaultdict

input_base_dir = '/home/hutr/iot-longitudinal/output_destination/all_devices_remove_duplicates'
input_base_dir = '/home/hutr/iot-longitudinal/output_destination/common_devices'
years = ['2019', '2021', '2022', '2023', '2024']
exps = ['idle', 'activity', 'power']
total_dict = defaultdict(int)
untranslated_dict = defaultdict(int)
for year in years:
    for exp in exps:
        input_dir = f'{input_base_dir}/{year}{exp}/ip_list'
        df = pd.read_csv(f'{input_dir}/_untranslated_ip_stats_with_all_dns.csv')
        # extract the last two numbers of last row: Total,56.1034275405893,933,1663
        total = df.iloc[-1, -1]
        untranslated_ips = df.iloc[-1, -2]
        total_dict[year] += total
        untranslated_dict[year] += untranslated_ips
        print(f"{year} {exp}", total, untranslated_ips)
    print(f"{year}:", total_dict[year], untranslated_dict[year])
print("Total:", sum(total_dict.values()))
print("Untranslated:", sum(untranslated_dict.values()))
print("Untranslated ratio:", sum(untranslated_dict.values()) / sum(total_dict.values()))

2019 idle 563 14
2019 activity 716 148
2019 power 499 16
2019: 1778 178
2021 idle 3097 601
2021 activity 591 257
2021 power 72 9
2021: 3760 867
2022 idle 4204 70
2022 activity 355 98
2022 power 47 0
2022: 4606 168
2023 idle 2306 15
2023 activity 258 25
2023 power 902 17
2023: 3466 57
2024 idle 2787 13
2024 activity 82 10
2024 power 412 125
2024: 3281 148
Total: 16891
Untranslated: 1418
Untranslated ratio: 0.08395003256171926


### Common_devices:
2019 idle 563 14
2019 activity 716 148
2019 power 499 16
2019: 1778 178
2021 idle 3097 601
2021 activity 591 257
2021 power 72 9
2021: 3760 867
2022 idle 4204 73
2022 activity 355 98
2022 power 47 0
2022: 4606 171
2023 idle 2306 16
2023 activity 258 25
2023 power 902 22
2023: 3466 63
2024 idle 2787 13
2024 activity 82 10
2024 power 412 140
2024: 3281 163
Total: 16891
Untranslated: 1442
Untranslated ratio: 0.08537090758392044

### All devices:
2019 idle 1423 144
2019 activity 1142 196
2019 power 902 42
2019: 3467 382
2021 idle 12514 1234
2021 activity 1156 342
2021 power 143 41
2021: 13813 1617
2022 idle 17631 269
2022 activity 1852 394
2022 power 251 67
2022: 19734 730
2023 idle 16366 432
2023 activity 693 83
2023 power 3362 100
2023: 20421 615
2024 idle 17481 154
2024 activity 430 63
2024 power 1604 593
2024: 19515 810
Total: 76950
Untranslated: 4154
Untranslated ratio: 0.05398310591293048